src/renderers/shared/utils/ReactErrorUtils.js

```diff
@@ -22,14 +22,14 @@ var caughtError = null;
  * @param {*} a First argument
  * @param {*} b Second argument
  */
-function invokeGuardedCallback<A, B, C>(
+function invokeGuardedCallback<A, B>(
   name: string,
-  func: (a: A, b: B) => C,
+  func: (a: A, b: B) => void,
   a: A,
   b: B,
-): ?C {
+): void {
   try {
-    return func(a, b);
+    func(a, b);
```

Comment:
you should probably remove the `return undefined;` line below, which is pretty much useless anyway